hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieCleaner.java

```diff
@@ -64,6 +64,13 @@ public HoodieCleaner(Config cfg, JavaSparkContext jssc) {
     LOG.info("Creating Cleaner with configs : " + props.toString());
   }
 
+  public HoodieCleaner(Config cfg, TypedProperties props, JavaSparkContext jssc) {
+    this.cfg = cfg;
+    this.jssc = jssc;
+    this.props = props;
```

Comment:
The original code fetches the props either from cmd line or global options, what is the purpose to support custom props per-service then?

Reply:
Yes in multi-table services we also fetch the props either from cmd line or global options. And we do not want each service to fetch the props by themselves. So here I add a new constructor to use the props that all services shared.

Reply:
> n multi-table services we also fetch the props either from cmd line or global options

The existing constructor already satisfy the needs.

Reply:
> The existing constructor already satisfy the needs.

Yes, however if we want to reuse existing constructor, we need pass the prop file path and the cmd line conifig from `HoodieMultiTableServicesMain` to `HoodieCleaner`. And we will parse the config for every `TableServiceTask` , which is redundant. Current implementation will do the config parsing in `HoodieMultiTableServicesMain` only, and all the `TableServiceTask` will use the final config parsed by `HoodieMultiTableServicesMain`. What do you think?

Reply:
Fine with it.

Reply:
Then we can make one constructor to invokes the other one.

Reply:
> Then we can make one constructor to invokes the other one.

Thanks for your suggestions~ I have slightly refactored the code and make it cleaner, PTAL